### DISCUSSION TITLE
How to force istio companion proxy to go to opentelemetry not using ssl?

### DISCUSSION THREAD

**Discussion by lserranov72** [istio/istio#49406] [General]
I have 2 pods, both use istio proxy. In one case, istio goes to opentelemetry not using ssl, in the other case ssl is used. 

What do I need to do to force istio to not to use ssl always?   I see in istio mesh something like below but I do not see how to control the use of ssl or not:

    defaultProviders:
      tracing:
        - opentelemetry
    extensionProviders:
      - name: "opentelemetry"
        opentelemetry:
          service: "opentelemetry-collector.istio-system.svc.cluster.local"
          port: 4317
    enablePrometheusMerge: true
    rootNamespace: null
    trustDomain: cluster.local

Thanks!

**Comment by zirain** [comment]
assume OTel-collector pod is not injected, envoy should always not sent ssl, can you share the config dump the pod which send with ssl.
run `istioctl pc all <pod_name>.<pod_namespace> -o yaml`

**Reply by lserranov72** [reply]
Thanks for your answer. 
I investigated and I may know the reason for the behavior where one pod comes secure and one non secure to open telemetry.  
The pod that comes non secure to open telemetry has Workload mTLS mode PERMISSIVE. 
The pod that comes secure to open telemetry   has Workload mTLS mode STRICT 

I do not think I can change out from PERMISSIVE, so maybe that means that I should configure opentelemetry to support TLS?  Or maybe if I get open telemetry injected that may help?

**Reply by zirain** [reply]
I think so, you can give a try.
